pdns/recursordist/pdns_recursor.cc

```diff
@@ -2285,13 +2283,18 @@ static string* doProcessUDPQuestion(const std::string& question, const ComboAddr
               eventTrace.add(RecEventTrace::LuaGetTag, ctag, false);
             }
           }
+          catch (const MOADNSException& moadnsexception) {
```

Comment:
> But do not ratelimit them. Note that `MOADNSExceptions` is derived from `std::runtime_errror` and not from `std::exception`.

Something is weird here, because  `std::runtime_error` derives from `std::exception`.

Reply:
I misread the docs on https://en.cppreference.com/w/cpp/error/runtime_error. The declaration does not show the base class, only the Inheritance diagram does.